### PR TITLE
#162171037 user reading stats 

### PR DESCRIPTION
**Pull request by el-Joft**
#### What does this PR do?
This PR allows users to create reading statistics
#### Description of Task to be completed?
- The User statistics are created based on the numbers of articles the User views. 
- The User Stats is created when the User access the get article route
> /api/articles/{slug}
- To get the User stat(s)
> /api/users/readerstats
#### How should this be manually tested?
- npm test
#### What are the relevant pivotal tracker stories?
[#162171037](https://www.pivotaltracker.com/story/show/162171037)

### CODE REVIEW

File: lib/ReadingStatsModelQuery.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+/**
+   *
+   * method to store reading statistics views
+   * @static
+   * @param {integer} readInfo
+   * @returns {json} return json object to user
+   * @memberof ReadingStats
+   */
+  static async createReaderStats(readInfo) {
+    const { articleId, userId } = readInfo;
```
Comment (thebere) [inline lib/ReadingStatsModelQuery.js:19]: No try catch here?

File: lib/ReadingStatsModelQuery.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+  /**
+   * @description { A function that the details of an article from the database }
+   * @param { object } userId
+   * @param { object } res
+   * @returns { object } getArticle
+   */
+  static async getUserReadingStats(userId) {
+    const user = await readerStats.findAndCountAll({
+      where: {
```
Comment (thebere) [inline lib/ReadingStatsModelQuery.js:45]: Missing try catch

File: middlewares/checkAuthentication.js [hunk 3]
```diff
@@ -6,6 +6,12 @@ import UserModelQuery from '../lib/UserModelQuery';
 dotenv.config();
 const checkAuthentication = async (req, res, next) => {
   const token = req.headers['access-token'];
+  // const articleUrl = () => {
```
Comment (thebere) [inline middlewares/checkAuthentication.js:9]: No commented out code

File: lib/ReadingStatsModelQuery.js (hunk 1 above)
Comment (el-Joft) [reply lib/ReadingStatsModelQuery.js:19]: I called the method in the ReadingStatController which has a try and catch. I thought even though there is an error the ReadingStatController catch will 'catch' the error. But I will like to get more understanding of it. Thanks

File: middlewares/checkAuthentication.js (hunk 3 above)
Comment (el-Joft) [reply middlewares/checkAuthentication.js:9]: Thanks. I will fix it

**Review by thebere — APPROVED** [review]

**Pull request merged**